README
Copyright (C) 2003 W. Michael Petullo <mike@flyn.org>

dmapd is Serve DAAP (iTunes) and DPAP (iPhoto) shares from Linux

= OVERVIEW =================================================================
	
The dmapd project provides a GObject-based, Open Source DMAP server 
with the following features:

 o Support for both DAAP and DPAP

 o Support for realtime transcoding of media formats not natively 
 supported by clients

 o Support for many metadata formats, such as those associated with Ogg 
 Vorbis and MP3 (e.g., ID3)

 o Detection of video streams so that clients may play them as video

 o Use of GStreamer to support a wide range of audio and video CODECs

 o Caching of photograph thumbnails to avoid regenerating them each time 
 the server restarts

Dmapd runs on Linux and other POSIX operating systems. It has been 
used on OpenWrt Linux-based systems with as little as 32MB of memory 
to serve music, video and photograph libraries containing thousands of 
files.

= BUILDING =================================================================

To build, cross your fingers and try...

 1. ./configure
 2. make
 3. make install

Read the ``INSTALL'' file for generic detailed information on installing
this program.
	
= NASTY DETAILS ============================================================
	
Dmapd can provide content to any client that supports DAAP or DPAP. 
This includes the following software clients and hardware devices:

 o Apple iTunes

 o Apple iPhoto

 o Rhythmbox

 o Roku SoundBridge

Dmapd can read metadata from any music file supported by GStreamer. 
In order to use this feature you must have the appropriate GStreamer 
plugins installed. The following plugins are always required if you 
wish to use this feature:

 o app

 o decodebin

 o typefindfunctions

The following GStreamer plugins are required for the corresponding 
media types:

 o mad (MP3)

 o id3demux (MP3)

 o ogg (Ogg Vorbis, etc.)

 o vorbis (Ogg Vorbis)

 o flac (FLAC)

 o mpeg2dec (MPEG video)

 o theora (Ogg Theora video)



ChangeLog
commit f800147c2919f151df811e745dfb3db00ad15681
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Feb 26 20:31:45 2011 -0600

    Bump version number
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 43d47049eb6318bada07e4172daabbcf377d99e5
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Feb 26 20:30:09 2011 -0600

    Make lockfile error message a little more clear
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 1018ba7ba774df97576b1a0a252a44421e790428
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Feb 26 19:26:41 2011 -0600

    Handle default DB plugin properly again
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 82b8e163f6ac3858c8fd08f3e2d522fb6630cd36
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Feb 26 19:24:42 2011 -0600

    Handle missing plugins more gracefully
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit b1e8a18c0f25dfb1adfaed2047c91a48a89727e3
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Feb 26 17:28:26 2011 -0600

    Print usage and quit if music and picture directory arguments not provided
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 6d30971c78793d8a6f11074790b54efd59cae672
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Feb 25 12:41:03 2011 -0600

    Two small fixes
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 75704c9814504a45d27e9a7da17a21a1a3ed248f
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Feb 4 12:42:05 2011 -0600

    Test new web-based Git feature
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 6c7df22ea7f1f705fdf227a6c5575f976c989fa1
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Feb 4 09:20:02 2011 -0600

    Get rid of some memory leaks identified by valgrind
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 5a70b0ecc87c4533b31d1de35582b26b5095eb5d
Author: W. Michael Petullo <mike@flyn.org>
Date:   Thu Feb 3 21:27:18 2011 -0600

    Fix create_connection prototype
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit a0786843325eabdf379d5a3898311835a5e40e92
Author: W. Michael Petullo <mike@flyn.org>
Date:   Thu Feb 3 21:26:18 2011 -0600

    Modify prototype of function to create connection
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 17396c67cde9b1ad144db717265fdd4b9ec1d966
Author: W. Michael Petullo <mike@flyn.org>
Date:   Thu Feb 3 21:24:51 2011 -0600

    Add dmapd stress test utility
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 92c9f48efd54292e6b2d2c6d4de7a2fd9b9b60e4
Author: W. Michael Petullo <mike@flyn.org>
Date:   Thu Feb 3 09:20:25 2011 -0600

    Add documentation of release protocol
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit ee6a2b6c040f3984c3e4296ea7f6765c09ca4605
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Jan 28 17:28:11 2011 -0600

    Bump version number
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit b37ccc27cf6a46773b66d8fc1b305ee9be2e2577
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Jan 28 16:09:39 2011 -0600

    Finish removing real_format from DmapdDAAPRecord
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 713b31f98d3b35aaa13a62cc7437b3f473d78bee
Author: W. Michael Petullo <mike@flyn.org>
Date:   Wed Jan 26 21:42:49 2011 -0600

    Add Noah to AUTHORS
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 9e911bffd12709b7bc9b9d505e8a2ab4de374bc9
Author: W. Michael Petullo <mike@flyn.org>
Date:   Mon Jan 24 16:59:19 2011 -0600

    Assorted updates
    
    Remove "real-format" property that is no longer used
    Modify to use libdmapsharing's new GByteArray-based thumbnail
    Update transcode code
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 3bdf4d6e96e8f49a8c768241037a620a86a2d274
Author: W. Michael Petullo <mike@flyn.org>
Date:   Mon Jan 3 22:20:50 2011 -0600

    Uncomment free of path in db_builder_gdir_build_db_starting_at (), seems okay
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 5b96ad8baf5625c476f981f8e49c17672e50dc3f
Author: W. Michael Petullo <mike@flyn.org>
Date:   Mon Jan 3 21:58:01 2011 -0600

    Require VIPS >= 7.24.1 for EXIF thumbnail and progressive JPEG detection
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 74f4f9859dee643d4cdef8969c9ddfb85037e1b9
Author: W. Michael Petullo <mike@flyn.org>
Date:   Mon Jan 3 21:54:22 2011 -0600

    Use VIPS' new disc-based decompression instead coding it here
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 7135edee36e246f5a47f3d687a51ba56cd59ffc2
Author: W. Michael Petullo <mike@flyn.org>
Date:   Thu Dec 23 12:25:03 2010 -0600

    Continue new libdmapsharing API work; allow non-realtime transcoding to database directory
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 29ee0fd4de21b09c5722a0015dbf99b1bacff2ca
Author: W. Michael Petullo <mike@flyn.org>
Date:   Tue Dec 14 10:08:42 2010 -0600

    Start migrating to new, Vala-friendly libdmapsharing API
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 269b8928e736004c18c72f3f9149320f6b66d99c
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Dec 11 16:48:00 2010 -0600

    Port unit tests to use check library
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 4865f03c5bd6983f360d0b597841f56604d07867
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sat Dec 4 21:49:10 2010 -0600

    Add ability to read EXIF thumbnails to VIPS code (requires VIPS 7.24)
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit 15dae21d3d8844204b245b4f5f687ccc5042cad9
Author: W. Michael Petullo <mike@flyn.org>
Date:   Fri Dec 3 09:17:44 2010 -0600

    Update documentation and some metadata-related code in the VIPS plugin
    Signed-off-by: W. Michael Petullo <mike@flyn.org>

commit e2e6d3dbd42a66b6793f3efdf65a54e2b00b2ed2
Author: W. Michael Petullo <mike@flyn.org>
Date:   Sun Nov 28 20:32:22 2010 -0600

    Bump version number

autogen.sh
#!/bin/sh

aclocal || exit 1
autoconf || exit 1
autoheader || exit 1
libtoolize --force || glibtoolize --force || exit 1
automake -a || exit 1
./configure $* || exit 1

git log > ChangeLog

cat <<EOF >> ChangeLog

======================== Convert to Git-based ChangeLog ========================

21 November 2010 W. Michael Petullo <mike@flyn.org>

	* Make thumbnail optional within DPAPRecord. I am having trouble
	with the memory use of libjpeg when decompressing multiscan
	JPEG's. I have modified my copy of VIPS to skip multiscan
	files. In this case, the resulting DPAPRecord does not have
	a thumbnail, but memory usage is managable on small systems
	(e.g., 32MB memory). I am presently working with the VIPS
	developers to resolve this.

19 November 2010 W. Michael Petullo <mike@flyn.org>

	* Reduce maximum size of VIPS in memory image manipulation from
	1MB to 512KB.

	* Don't store location in DmapdDMAPDbDisk's db (location is
	still in the actual DPAP records, of course), reducing memory
	usage by 200KB when sharing a 4,000-photograph database with a
	reasonable database directory path.

	* Fix crash on OpenWrt by commenting out VIPS EXIF comment code.

18 November 2010 W. Michael Petullo <mike@flyn.org>

	* Fix minor memory issues identified by valgrind.

15 November 2010 W. Michael Petullo <mike@flyn.org>

	* Fix compile warnings.

	* Port test code to new modular database backends.

10 November 2010 W. Michael Petullo <mike@flyn.org>

	* Ensure modules are never loaded twice.

06 November 2010 W. Michael Petullo <mike@flyn.org>

	* Work on BDB module: Apply iPhoto G_MAXUINT vs. G_MAXINT (see
	previous for GHashTable DB); fix double-free of DPAP records'
	stringletons.

04 November 2010 W. Michael Petullo <mike@flyn.org>

	* Move from libdb-4.7 to libdb-48.

	* Begin modularizing backend DB code.

17 October 2010 W. Michael Petullo <mike@flyn.org>

	* Delete temporary VIPS thumbnail files when done with them.

16 October 2010 W. Michael Petullo <mike@flyn.org>

	* Fix missing NULL terminators for g_object_set(..."mediakind"...)

15 October 2010 W. Michael Petullo <mike@flyn.org>

	* Allow configuring of group (was hard-coded to "daemon").

30 September 2010 W. Michael Petullo <mike@flyn.org>

	* Implement suggestions from Maxim Bourmistrov: clean up default
	module code, add example to dmapd.conf, replace g_memdup and
	disallow both DAAP and DPAP data written to same cache file.

26 September 2010 W. Michael Petullo <mike@flyn.org>

	* Use max int, not max unsigned int for first DMAPDb ID. iPhoto
	6 does not seem to be able to handle max unsigned int (was seeing
	ID=-x in requests).

20 September 2010 W. Michael Petullo <mike@flyn.org>

	* Warn, don't error out if a module is missing.

12 September 2010 W. Michael Petullo <mike@flyn.org>

	* Remove check that forbid serving pictures and music while
	using -f option.

10 September 2010 W. Michael Petullo <mike@flyn.org>

	* Fix failed assertion having to do with empty playlists.

09 September 2010 W. Michael Petullo <mike@flyn.org>

	* Do not fork a process for each DMAP sub-protocol.

09 September 2010 W. Michael Petullo <mike@flyn.org>

	* Check for NULL before unref'ing to fix segfault on OpenWrt.

07 September 2010 W. Michael Petullo <mike@flyn.org>

	* Use DMAPMediaKind enum.

	* Change mistaken gboolean for mediakind to gint.

06 September 2010 W. Michael Petullo <mike@flyn.org>

	* Support DAAPRecord's sort-album and sort-artist properties.

	* Fix serving to iTunes 10: support com.apple.itunes.mediakind.

23 August 2010 W. Michael Petullo <mike@flyn.org>

	* Fix a memory leak in VIPS usage found using valgrind.

22 August 2010 W. Michael Petullo <mike@flyn.org>

	* Fix crash due to trying to load non-image using image reader.

	* Fix thumbnailing code in VIPS plugin (aspect ratio MUST be valid).

19 August 2010 W. Michael Petullo <mike@flyn.org>

	* Fix identifying duration of media files (forgot to set
	requested format to GST_FORMAT_TIME; this variable is used
	for both input and output for/from gst_element_query_duration ()).

18 August 2010 W. Michael Petullo <mike@flyn.org>

	* Adopt songalbum over daap.songalbum for DAAP record properties
	in accordance with new libdmapsharing.

06 August 2010 W. Michael Petullo <mike@flyn.org>

	* Add systemd .service file.

22 June 2010 W. Michael Petullo <mike@flyn.org>

	* Make Berkeley DB backend optional. Will eventually turn the
	DB backends into modules that may be loaded at runtime.

03 June 2010 W. Michael Petullo <mike@flyn.org>

	* Do not strdup before returning a string property of a
	DmapdD[AP]ARecord.

02 June 2010 W. Michael Petullo <mike@flyn.org>

	* Implement stringletons -- string singletons -- to further
	reduce memory usage. One use case reduced heap from 3,346,432
	to 2,826,249 bytes.

29 May 2010 W. Michael Petullo <mike@flyn.org>
	
	* Implement persistent databases.

28 May 2010 W. Michael Petullo <mike@flyn.org>

	* Fix compiler warnings.

27 May 2010 W. Michael Petullo <mike@flyn.org>

	* Free avahi client when finalizing DMAPShare.

	* Fix prototype for signal handlers.

	* Fix several items that resulted in compiler warnings.

26 May 2010 W. Michael Petullo <mike@flyn.org>

	* Add finalize code to DmapdDMAPContainerRecord.

25 May 2010 W. Michael Petullo <mike@flyn.org>

	* Bad assertion fixes.

	* Fix some memory leaks identified by valgrind.

23 May 2010 W. Michael Petullo <mike@flyn.org>

	* Fix dmapd_dmap_container_db_foreach, I don't know why I was
	using GFunc instead of GHFunc.

22 May 2010 W. Michael Petullo <mike@flyn.org>

	* Berkeley Database work.

	* Add db_dir configuration option.

20 May 2010 W. Michael Petullo <mike@flyn.org>

	* Add finalize code for:
		DmapdDMAPContainerDb
		DmapdDMAPDbBDB
		DmapdDMAPDbGHashTable

	* Free GObjectContext * in main.

	* Fix debug statement of has_video.

	* Use gst_object_unref instead of g_object_unref when appropriate.

18 May 2010 W. Michael Petullo <mike@flyn.org>

	* Do not try to set options if /etc/dmapd.conf does not exist.

	* Work on VIPS backend. This should allow thumbnail generation
	on low-memory systems.

17 May 2010 W. Michael Petullo <mike@flyn.org>

	* Continued to work memory usage issues. Found a few small and
	one larger memory leak in av-meta-reader-gst.c. The large leak
	is troublesome because its fix is to free a const gchar *. See
	FIXME comments is source file.

16 May 2010 W. Michael Petullo <mike@flyn.org>

	* Added Berkeley Database backend while trying to get to the
	bottom of memory usage. There seems to be a memory leak caused
	by av_meta_reader_gst_read, so more work is needed.

17 February 2010 W. Michael Petullo <mike@flyn.org>

	* Change dmapd.conf to set user by default.

05 February 2010 W. Michael Petullo <mike@flyn.org>

	* Parent kills children when SIGTERM is received.

	* Demote warning about unused metadata to a debug message.

31 January 2010 W. Michael Petullo <mike@flyn.org>

	* Add OpenWRT init script.

	* Various file cleanups.

	* Fix crash when a user provides a bad module path (e.g., to
	DMAPD_PHOTO_MODULE).

28 January 2010 W. Michael Petullo <mike@flyn.org>

	* Use lockfile=/var/lock/subsys/dmapd in init script to appease
	rpmlint (used /var/lock/subsys/$prog before).

	* Create default configuration file.

16 January 2010 W. Michael Petullo <mike@flyn.org>

	* Add configuration file support.

	* Provide more descriptive error if a media directory can't be read.

17 December 2009 W. Michael Petullo <mike@flyn.org>

	* Add vips plugin.

12 December 2009 W. Michael Petullo <mike@flyn.org>

	* Don't segfault if a GStreamer plugin is not available while
	reading AV metadata.

05 December 2009 W. Michael Petullo <mike@flyn.org>

	* Properly use new libdmapsharing log domain.

24 November 2009 W. Michael Petullo <mike@flyn.org>

	* Allow the user to set meta reader modules to "null" if he does
	not want to load any module.

22 November 2009 W. Michael Petullo <mike@flyn.org>

	* Move default data dir from /var/db/Storage to /var/db/dmapd.

	* Add option to drop root privileges.

10 November 2009 W. Michael Petullo <mike@flyn.org>

	* Set library version properly.

28 August 2009 W. Michael Petullo <mike@flyn.org>

	* Remove set_transcode function.

17 August 2009 W. Michael Petullo <mike@flyn.org>

	* Ensure dmapd-d[ap]ap-record.h is put in tarball.

13 August 2009 W. Michael Petullo <mike@flyn.org>

	* Proper reference counting of DMAPRecords in DMAPDbs.

10 August 2009 W. Michael Petullo <mike@flyn.org>

	* Reduce the number of installed header files.

08 August 2009 W. Michael Petullo <mike@flyn.org>

	* Fix issues related to building a module external to the dmapd
	source tree.

07 August 2009 W. Michael Petullo <mike@flyn.org>

	* Fix crash in new GTypeModule photo code.

06 August 2009 W. Michael Petullo <mike@flyn.org>

	* Complete new GTypeModule module work.

04 August 2009 W. Michael Petullo <mike@flyn.org>

	* Port DbBuilder to new API.

02 August 2009 W. Michael Petullo <mike@flyn.org>

	* Just include <libdmapdsharing/dmap.h>

	* Began to refactor plugin code.

31 July 2009 W. Michael Petullo <mike@flyn.org>

	* Use g_module_build_path instead of g_strconcat.

	* New module GObjects.

24 July 2009 W. Michael Petullo <mike@flyn.org>

	* Don't create versioned module libraries.

22 July 2009 W. Michael Petullo <mike@flyn.org>

	* Fix warning: DmapdD[AP]APRecord now implements DMAPRecord
	and D[AP]APRecord.

	* Fixed various runtime warnings.

21 July 2009 W. Michael Petullo <mike@flyn.org>

	* Fix handling of DAAP filesize.

20 July 2009 W. Michael Petullo <mike@flyn.org>

	* Handle errors when reading metadata; don't add to database.

19 July 2009 W. Michael Petullo <mike@flyn.org>

	* Make path to lock- and pidfile configurable at runtime.

13 July 2009 W. Michael Petullo <mike@flyn.org>

	* Move libmeta-gst.so to libav-meta-gst.so.

	* Create libphoto-meta-graphicsmagick.so.

10 July 2009 W. Michael Petullo <mike@flyn.org>

	* Create libdb-gdir.so and libmeta-gst.so modules.

05 July 2009 W. Michael Petullo <mike@flyn.org>

	* Move transcoding to libdmapsharing.

02 July 2009 W. Michael Petullo <mike@flyn.org>

	* Get rid of get methods in dmapd-daap-record.c and replace with
	GObject properties.

28 June 2009 W. Michael Petullo <mike@flyn.org>

	* Continue work to delay determining transcode format.

17 June 2009 W. Michael Petullo <mike@flyn.org>

	* Delay determining transcode format.

11 June 2009 W. Michael Petullo <mike@flyn.org>

	* Began to set the conditions for pulling GStreamer code into
	a dynamic module.

10 June 2009 W. Michael Petullo <mike@flyn.org>

	* Started to make DAAP or DPAP support optional.

01 May 2009 W. Michael Petullo <mike@flyn.org>

	* Increased the time waited for the decoded buffer to have room
	before dropping transcoded data in an effort to fix
	skipping. Previously audio would skip when decoding outpaced
	the client's ability to read from the network (especially
	prevalent when transcoding not necessary, making GStreamer's
	work very easy).

30 April 2009 W. Michael Petullo <mike@flyn.org>

	* Fix compiler warnings.

23 April 2009 W. Michael Petullo <mike@flyn.org>

	* Write GGst[Format]InputStreamFactory classes.

19 April 2009 W. Michael Petullo <mike@flyn.org>

	* Fix segfault when loading audio file that GStreamer can not
	recognize.

	* Fix segfault when loading picture that GraphicsMagick can not
	recognize.

	* Create subclasses of GGstInputStream providing for MP3,
	WAV and raw (original encoding) streams.

18 April 2009 W. Michael Petullo <mike@flyn.org>

	* Start work on user configurable transcoding format

10 April 2009 W. Michael Petullo <mike@flyn.org>

	* dmapd.c not #include's unistd.h to fix Mac OS X build.

	* Transcode using a depth and width of 8 (instead of 16) to save
	network bandwidth.

07 April 2009 W. Michael Petullo <mike@flyn.org>

	* Change to reflect libdmapsharing's use of a guint for record
	ID's.

29 March 2009 W. Michael Petullo <mike@flyn.org>

	* Fix a lot of compiler warnings.

	* Check for existence of /var/cache/dmapd.

26 March 2009 W. Michael Petullo <mike@flyn.org>

	* Fix determining format from real_format.

	* Attempt to fix a deadlock that can occur when
	g_gst_input_stream_close () is called.

24 March 2009 W. Michael Petullo <mike@flyn.org>

	* Make dmapd_dmap_container_record_get_entries return a const
	DMAPDb *.

21 March 2009 W. Michael Petullo <mike@flyn.org>

	* Limit decoded audio buffer size so that it does not exhaust
	memory to store an uncompressed song.

	* Start work on seeking transcoded streams.

20 March 2009 W. Michael Petullo <mike@flyn.org>

	* Transcode to WAV instead of MP3 to satisfy Fedora.

14 March 2009 W. Michael Petullo <mike@flyn.org>

	* Make GGstInputStream implement GSeekable.

01 March 2009 W. Michael Petullo <mike@flyn.org>

	* Fix rare timing issue that caused buffer overflow.

22 February 2009 W. Michael Petullo <mike@flyn.org>

	* Work to clean up g-gst-input-stream.c and dmapd-daap-record.c.

13 February 2009 W. Michael Petullo <mike@flyn.org>

	* Add support for FLAC.

12 February 2009 W. Michael Petullo <mike@flyn.org>

	* Migrated to new libdmapsharing API. Support Roku SoundBridge
	line.

09 February 2009 W. Michael Petullo <mike@flyn.org>

	* Fixed media type detection for audio / video.

01 February 2009 W. Michael Petullo <mike@flyn.org>

	* Implement realtime transcoding.

	* Create media tag reader pipeline only once.

	* Determine format by looking at stream.

30 January 2009 W. Michael Petullo <mike@flyn.org>

	* Fix some memory leaks.

29 January 2009 W. Michael Petullo <mike@flyn.org>

	* Read metadata from audio / video files.

28 January 2009 W. Michael Petullo <mike@flyn.org>

	* Add detection of video CODEC and set has_video.

	* Fix double call to DMAP factory create.

21 January 2009 W. Michael Petullo <mike@flyn.org>

	* Add -n, set share name.

	* Wait for child processes properly.

19 January 2009 W. Michael Petullo <mike@flyn.org>

	* Add support for has-video meta record.

	* Modify to use a record factory.

	* Allow multiple media directory roots.

	* Serve DAAP and DPAP simultaneously.

	* Work on init script.

18 January 2009 W. Michael Petullo <mike@flyn.org>

	* Modified to reflect changes to libdmapsharing API.

16 January 2009 W. Michael Petullo <mike@flyn.org>

	* Add a photo data disk cache so that metadata does not need to
	be re-read / re-generated each time you run dmapd.

	* Fork into a daemon.

	* Media ID's start at maximum and go down. Container ID's start
	at 1 and go up.

11 January 2009 W. Michael Petullo <mike@flyn.org>

	* Fix aspect ratio of thumbnails for photos that have a portrait
	orientation.

10 January 2009 W. Michael Petullo <mike@flyn.org>

	* Added album support.

28 December 2008 W. Michael Petullo <mike@flyn.org>

	* Began project.
EOF

exit 0
